civil.go

```diff
@@ -49,16 +49,26 @@ func DateOf(t time.Time) Date {
 }
 
 // RFC3339Date is the civil date format of RFC3339
-const RFC3339Date = "2006-01-02"
+const (
+	RFC3339Date       = "2006-01-02"
+	RFC3339DateNoDash = "20060102"
+)
 
 // ParseDate parses a string in RFC3339 full-date format and returns the date value it represents.
 func ParseDate(s string) (Date, error) {
-	const dateZero = "0000-00-00"
+	return parseDate(s, RFC3339Date, "0000-00-00")
+}
+
+// ParseDateNoDash parses a string in RFC3339 full-date format, without any dashes, and returns the date value it represents.
+func ParseDateNoDash(s string) (Date, error) {
```

Comment:
if your requirement that you will know ahead of time, if the string representation of a date you get has or doesn't have dashes?

Reply:
Correct, the caller must know which format they're dealing with.

Reply:
ok, now, just for curiosity, what would be the drawback of having a single parse function that would try one format and if it fails, tries the second format? (not blocking the PR on this question, just trying to have a dialog around it)

Reply:
Oh for sure! 
1. The added overhead and complexity in the underlying function, though not much, still exists and makes the change riskier. The expected cost has now changed for all callers.
2. If the function accepts both formats, the data could potentially change underneath the caller without anyone noticing. While not inherently bad, something that would have been expected to fail before now does not which could lead to other failures and logic needing adjusting across the board (depending on how often this was being used and where)
3. The principal of explicitly handling an expected value with a given format is now ambiguous, and can no longer be relied on.
4. We would now be (potentially) setting a precedent that changing how types handle incoming data could be more dynamic and would potentially embolden other developers to make more complex changes. (maybe not, but there would a reference point to where it would be done)

Reply:
Piggy backing on more of the principal of being explicit, we could also potentially expose the private method and make it public so if anyone did need to define their own one off layout could do it as well without having to make a change to this lib.

Reply:
thanks you for your reply! all good points.